pkg/identity/identitysync/sync.go

```diff
@@ -18,16 +18,16 @@ import (
 )
 
 type IdentityProvider interface {
-	Name() string
```

Comment:
@chrnorm FYI we removed this from the interface and instead get this context from the IdentitySyncer struct after initialisation.

This makes it simpler to implement groups source and avoids a class of issue where groups returned by an idp have different sources or we use the wrong idptype IDs in the code.

We found that the Name() methods were returning different IDs than the IDPType constants. Was this intentional? and is it a breaking change for analytics?

Reply:
Not a breaking change for analytics! The inconsistency here was just my personal preference (in hindsight I preferred “onelogin” instead of “one-login”, “azure-ad” instead of “azure”). Happy to change these back to be consistent with the YAML config to keep things simple and consistent - so please remove the inconsistencies I’ve introduced.